src/AmqpBundle/Amqp/Producer.php

```diff
@@ -51,6 +52,17 @@ public function publishMessage($message, $flags = AMQP_NOPARAM, array $attribute
                       array_merge($this->exchangeOptions['publish_attributes'], $attributes));
 
         $routingKeys = !empty($routingKeys) ? $routingKeys : $this->exchangeOptions['routing_keys'];
+
+        if ($this->eventDispatcher) {
```

Comment:
@b-viguier I actually tried to implement the NullEventDispatcher but I gave up since you'd have to inject it in the construct of the abstract class and it would become a BC. I prefer to leave it as it is (an optional dependency).